doc/_templates/footer.html

```diff
@@ -3,7 +3,7 @@
 
 {%- block extrafooter %}
 <p>
-Built with <a href="https://sphinx-doc.org/">Sphinx</a>
+Built with <a href="https://www.sphinx-doc.org/en/master/">Sphinx</a>
```

Comment:
I prefer to omit the "en/master" part.
```suggestion
Built with <a href="https://www.sphinx-doc.org/">Sphinx</a>
```